crates/fda/src/main.rs

```diff
@@ -1159,20 +1209,56 @@ async fn pipeline(format: OutputFormat, action: PipelineAction, client: Client)
                 ))
                 .unwrap();
 
-            let mut byte_stream = response.into_inner();
-            while let Some(chunk) = byte_stream.next().await {
-                let mut buffer = Vec::new();
-                match chunk {
-                    Ok(chunk) => buffer.extend_from_slice(&chunk),
-                    Err(e) => {
-                        eprintln!("ERROR: Unable to read server response: {}", e);
-                        std::process::exit(1);
+            match format {
+                OutputFormat::Text | OutputFormat::Json => {
+                    let mut byte_stream = response.into_inner();
+                    while let Some(chunk) = byte_stream.next().await {
+                        let mut buffer = Vec::new();
+                        match chunk {
+                            Ok(chunk) => buffer.extend_from_slice(&chunk),
+                            Err(e) => {
+                                eprintln!("ERROR: Unable to read server response: {}", e);
+                                std::process::exit(1);
```

Comment:
exit?
isn't this a little too dramatic?
Shouldn't this return an error and the exist should be higher up?

Reply:
this is in the CLI tool there arent many levels